.github/workflows/changelog.yml

```diff
@@ -1,34 +1,35 @@
-name: Check Changelog.md
+name: Check PR for Changelog
+
 on:
   pull_request:
-    types: [opened, synchronize, reopened]
+    types: [opened, synchronize, edited, reopened]
+
 jobs:
-  check-changelog:
-     runs-on: ubuntu-latest
-     permissions: write-all
-     steps:
-          - uses: actions/checkout@v4
-          - name: Fetch changed files
-            id: fetch-changed-files
-            run: |
-              response=$(gh api \
-              -H "Accept: application/vnd.github+json" \
-              -H "X-GitHub-Api-Version: 2022-11-28" \
-               repos/${owner}/${REPO}/pulls/${PULL_NUMBER}/files)
-              echo "$response" | jq '.' > response.json
-              
-          
-            env:
-              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-              owner: ${{ github.repository_owner }}
-              REPO: ${{ github.event.repository.name }}
-              PULL_NUMBER: ${{ github.event.pull_request.number }}
-          - name: Check if Changelog.md is changed
-            run: |
-              if [[ $(jq '.[].filename' response.json) == *"CHANGELOG.md"* ]]; then
-                echo "Changelog.md is changed"
-                exit 0
-              else
-                echo "Changelog.md is not changed"
-                exit 1
-              fi
+  check-description:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout code
```

Comment:
It seems that we dont need to checkout the code as we are not going to do any operations with the PRs code but with its metadata, so we dont need to checkout the branch for that. I just tested this without this step and the action keeps working fine.

Reply:
I will leave this for @rtibbles's consideration